tests/Computation.Tests/Values/AndTests.cs

```diff
@@ -14,74 +11,36 @@ public class AndTests
     [Theory]
     [ClassData(typeof(BinaryTestData))]
     private void ShouldCreateEquivalentConstants(
-        IConstantValue constantLeft, IConstantValue constantRight,
-        SymbolicUnsigned symbolicLeft, SymbolicUnsigned symbolicRight)
+        IValue left0, IValue right0,
+        IValue left1, IValue right1)
     {
-        var constant = And.Create(constantLeft, constantRight).AsConstant(Context);
-        var symbolic = And.Create(symbolicLeft, symbolicRight).AsConstant(Context);
+        var result0 = And.Create(left0, right0).AsConstant(Context);
+        var result1 = And.Create(left1, right1).AsConstant(Context);
 
-        constant.Should().Be(symbolic);
+        result0.Should().Be(result1);
     }
 
     [Theory]
     [ClassData(typeof(BinaryTestData))]
     private void ShouldCreateEquivalentBitVectors(
-        IConstantValue constantLeft, IConstantValue constantRight,
-        SymbolicUnsigned symbolicLeft, SymbolicUnsigned symbolicRight)
+        IValue left0, IValue right0,
+        IValue left1, IValue right1)
     {
-        var constant = And.Create(constantLeft, constantRight).AsBitVector(Context).Simplify();
-        var symbolic = And.Create(symbolicLeft, symbolicRight).AsBitVector(Context).Simplify();
+        var result0 = And.Create(left0, right0).AsBitVector(Context).Simplify();
+        var result1 = And.Create(left1, right1).AsBitVector(Context).Simplify();
 
-        constant.Should().BeEquivalentTo(symbolic);
+        result0.Should().BeEquivalentTo(result1);
     }
 
     [Theory]
     [ClassData(typeof(BinaryTestData))]
     private void ShouldCreateEquivalentBooleans(
-        IConstantValue constantLeft, IConstantValue constantRight,
-        SymbolicUnsigned symbolicLeft, SymbolicUnsigned symbolicRight)
+        IValue left0, IValue right0,
+        IValue left1, IValue right1)
     {
-        var constant = And.Create(constantLeft, constantRight).AsBool(Context).Simplify();
-        var symbolic = And.Create(symbolicLeft, symbolicRight).AsBool(Context).Simplify();
+        var result0 = And.Create(left0, right0).AsBool(Context).Simplify();
+        var result1 = And.Create(left1, right1).AsBool(Context).Simplify();
 
-        constant.Should().BeEquivalentTo(symbolic);
-    }
-
-    [Theory]
-    [ClassData(typeof(UnaryTestData))]
-    private void ShouldShortCircuitToZeroWhenEitherArgumentIsZero(
```

Comment:
It's not clear to me that the above tests cover this. I'm probably missing something, but my understanding is that the above tests assert that things are equal when going through the various symbolic and constant permutations (providing the symbolic actually contains a constant value). 

Whereas these tests are proving that some identity laws are held up. Isn't it still possible to break the implementation of these boolean operators for the identity cases but the above tests pass because the symbolic and constant paths both contain the same error?